### PR TITLE
Task2 babel+es6+

### PR DESCRIPTION
**Pull request by dzgweb**
Babel is connected as a part of build process, application works fine in all browsers from browser matrix including IE11.  Using Async / Await major features from ECMAScript 2016+

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +28,14 @@
...
     "gulp-uglify": "^3.0.1",
-    "gulp-util": "^3.0.8"
+    "gulp-util": "^3.0.8",
+    "nodelist-foreach-polyfill": "^1.2.0",
+    "webpack": "^4.26.0",
+    "webpack-dev-server": "^3.1.10"
+  },
+  "dependencies": {
+    "@babel/polyfill": "^7.0.0",
+    "webpack-cli": "^3.1.2",
```
Comment (NikitaSedyx) [inline package.json]: webpack is dev depth

File: package.json [hunk 2]
```diff
@@ -2,14 +2,22 @@
...
+    "gulp": "gulp",
+    "build": "babel app/js --out-file app/bundle.js",
+    "watch": "babel app/js --watch --out-file app/bundle.js",
+    "dev": "webpack-dev-server --inline --hot",
+    "webpack": "webpack"
   },
   "author": "Sergey Dziadevich",
   "license": "ISC",
   "devDependencies": {
+    "@babel/cli": "^7.1.5",
```
Comment (NikitaSedyx) [inline package.json]: remove `^`

File: package.json [hunk 3]
```diff
@@ -2,14 +2,22 @@
...
   "description": "ES6 training application that uses newsapi.org",
-  "main": "index.js",
+  "main": "webpack.config.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "gulp": "gulp"
+    "gulp": "gulp",
+    "build": "babel app/js --out-file app/bundle.js",
+    "watch": "babel app/js --watch --out-file app/bundle.js",
+    "dev": "webpack-dev-server --inline --hot",
```
Comment (NikitaSedyx) [inline package.json]: ideally gulp should run this

File: app/index.html [hunk 4]
```diff
@@ -17,6 +17,7 @@
   <link rel="icon" type="image/png" sizes="32x32" href="img/favicon/favicon-32x32.png">
   <link rel="icon" type="image/png" sizes="16x16" href="img/favicon/favicon-16x16.png">
 
+  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/css/bootstrap.min.css" integrity="sha384-MCw98/SFnGE8fJT3GXwEOngsV7Zt27NXFoaoApmYm81iuXoPkFOJwJ8ERdknLPMO" crossorigin="anonymous">
```
Comment (NikitaSedyx) [inline app/index.html]: without npm, it's non-production way

please investigate why

File: app/js/api.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+const apiKey = 'eabd967104da4e07a9c41b1342a889b1';
+const host = 'https://newsapi.org/v2';
+
+export class NewsApi{
+  static async getSourcesOnCategory(category){
+    const url = `${host}/sources?category=${category}&apiKey=${apiKey}`;
+    return fetch(url).then(response => response.json());
```
Comment (NikitaSedyx) [inline app/js/api.js]: `.then(response => response.json())` is copy paste

File: app/js/common.js [hunk 6]
```diff
@@ -1,12 +1,49 @@
...
-  let elem = elemId;
-  if (elem && news.status == 'ok') {
-    elem.innerHTML = news.articles
-      .map(a => '<li class="col-xs-12 col-sm-12 col-lg-6 col-xl-6">' + render(a) + '</li>')
-      .reduce((txt, li) => txt + li, '<ul class="articles row">') + '</ul>';
+const newCategories = ['business', 'entertainment', 'general', 'health', 'science', 'sports', 'technology'];
+const sourcesContainer = document.querySelector('#news-sources');
+const newsContainer = document.getElementById('news-container');
+
+let htmlSelect = `
```
Comment (NikitaSedyx) [inline app/js/common.js]: name doesn't tell me a lot

it's html, ok, it's select, and that's it

File: app/js/common.js [hunk 7]
```diff
@@ -1,12 +1,49 @@
...
+  <select class="form-control sel-cat" id="cat_val" name="cat-select">
+    ${newCategories.reduce((allOpt, opt) => allOpt + `<option value="${opt}">${opt}</option>`, `<option value="0">Select news category:</option>`)}
```
Comment (NikitaSedyx) [inline app/js/common.js]: long line

File: app/js/common.js [hunk 8]
```diff
@@ -1,12 +1,49 @@
...
   }
-}
+  getSource();
+});
+
+function renderSources(elemId, sources) {
+  let elem = elemId;
+  if (elem && sources.status == 'ok') {
+    elem.innerHTML = sources.sources
+      .reduce((txt, li) => txt + `<li class="col-xs-6 col-sm-6 col-lg-3 col-xl-3" data-sourceid='${li.id}'>${li.name}</li>`, `<ul class='row'>`) + '</ul>';
```
Comment (NikitaSedyx) [inline app/js/common.js]: long line

File: package.json (hunk 2 above)
Comment (dzgweb) [reply package.json]: remove character ^ or line of dependency?

**Pull request closed without merging**